caddyhttp/proxy/reverseproxy.go

```diff
@@ -254,6 +254,10 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 
 func copyHeader(dst, src http.Header) {
 	for k, vv := range src {
+		// some predefined headers (e.g. MIME plugin), skip.
+		if _, ok := dst[k]; ok {
+			continue
+		}
```

Comment:
@tw4452852 I've been looking into it too, but I ran out of time pretty quick so I asked @adamwathan and he says: https://twitter.com/adamwathan/status/779007715670499328

> nginx will use the mime that comes back from the proxy if it's there, but if not it adds the correct header itself
> I just figured that out now though, by passing an empty Content-Type header I can trigger nginx to do the look up for me.